timesketch/api/v1/resources/sigma.py

```diff
@@ -90,3 +93,39 @@ def get(self, rule_uuid):
                 HTTP_STATUS_CODE_NOT_FOUND, 'No sigma rule found with this ID.')
 
         return return_rule
+
+class SigmaByTextResource(resources.ResourceMixin, Resource):
+    """Resource to get list of Sigma rules."""
+
+    @login_required
+    def post(self):
+        """Handles POST request to the resource.
+
+        Returns:
+            JSON sigma rule
+        """
+        form = forms.SigmaRuleForm.build(request)
+        # optional: do we want to have a title at this point?
+        title = form.title.data
+        content = form.content.data
+
+        print(f'Content passed: {content}')
+        if not form.validate_on_submit():
+            abort(HTTP_STATUS_CODE_BAD_REQUEST, 'Unable to validate form data')
+
+        try:
+            sigma_rule = ts_sigma_lib.get_sigma_rule_by_text(content)
+
+        except ValueError:
+            # TODO: make better responses here with different scenarios.
+            logger.error('Parsing error',
+                         exc_info=True)
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
+                'Error unable to parse the provided Sigma rule')
+
+        if sigma_rule is None:
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND, 'No sigma was parsed')
+
+        return sigma_rule
```

Comment:
I'm seeing few things here...

a) I'm not seeing the correct structure returned, a dict with `meta` and `objects`... here should be a structure like that
b) you return back None if there are some issues... then you just check `is None` and have no meaningful error message, I think raising an exception is better, and catching that exception and using that in the return value is better, then the user in the UI/CLI/API will at least be able to see the error message that you get back. Instead of just `no sigma was parsed`, which doesn't tell you much (requiring a Timesketch SRE to look at logs to get some data back)

Reply:
ok the ```return None```aspect is fixed. The correct data structure indeed needs to be checked.